index.js

```diff
@@ -0,0 +1,143 @@
+            }
+        }
+        if(event.target.parentNode.className === "remove"){
+            let liToRemove = event.target.parentNode.parentNode;
+            for(let i = 0; i<todos.length; i++){
+                if(liToRemove.children[1].textContent===todos[i]){
+                    while(i<todos.length-1){
+                        todos[i] = todos[i+1];
+                        i++;
+                    }
+                    todos.pop();
+                }
+            }
+            liToRemove.style.opacity = "0";
+            setTimeout(()=>{list.removeChild(liToRemove);
+                        if(list.children.length===0){
+                        let li = document.createElement('LI');
+                        li.appendChild(document.createTextNode("you have no todos"));
+                        list.appendChild(li);}},400);
+        }
+    }
+    
+    let createLi = function(){
+        let todoButton = document.createElement('BUTTON');
+        todoButton.appendChild(document.createElement('IMG'));
+        todoButton.firstChild.src = "unticked.png";
+        todoButton.className = "isdone";
+        let rmvButton = document.createElement('BUTTON');
+        rmvButton.appendChild(document.createElement('IMG'));
+        rmvButton.firstChild.src = "remove.png";
+        rmvButton.className = "remove";
+        let span = document.createElement('SPAN');
+        span.appendChild(document.createTextNode(input.value));
+        let li = document.createElement('LI');
+        li.appendChild(todoButton);
+        li.appendChild(span);
+        li.appendChild(rmvButton);
+        li.className = "todoitem";
+        if(menu.children[1].className === "menu focused"){
+            li.style.display = 'none';
+        }
+        return li;
+    }
+
+    let addHandler = function(){
+        if(!(input.value=== ""||input.value == null)){
+            if(list.children[0].textContent === "you have no todos"){
+                list.removeChild(list.children[0]);
+            }
+            for(let i=0;i<todos.length;i++){
+                if(todos[i]===input.value){
+                    alert("you already have this task");
+                    return;
+                }
+            }
+            todos.push(input.value);
+            list.appendChild(createLi());
+            input.value = "";
+        }
+        else{
+            alert("no task to add");
+        }
+    }
+
+    let changeListMode = function(mode){
+        if(mode === "done"){
+            if(list.className === "list all"){
+                list.className = "allToDone";
+                setTimeout(()=>{list.className = "list done";},400);
+            }
+            else{
+                list.className = "list done";
+            }
+        }
+        else{
+            if(list.className === "list all"){
+                list.className = "allToUndone";
+                setTimeout(()=>{list.className = "list undone";},400);
```

Comment:
Note that you have some code duplication here (specifically here with the `setTimeout` expression).
You can move this out to a function, like this for instance:
```js
function animate(element, startClass, endClass) {
  element.classList.add(startClass);

  setTimeout(() => {
    element.classList.remove(startClass);
    element.classList.add(endClass);
  }, 400);
}
```